0.12.1 compilation error in lodestar

When building v0.12.1 with lodestar, I got the following error

```
$ tsc -p tsconfig.build.json
../../node_modules/libp2p-gossipsub/src/index.d.ts:106:5 - error TS2416: Property 'log' in type 'Gossipsub' is not assignable to the same property in base type 'PubsubBaseProtocol'.
  Type 'import("/Users/tuyennguyen/Projects/lodestar/node_modules/@types/debug/index").Debugger & { err: import("/Users/tuyennguyen/Projects/lodestar/node_modules/@types/debug/index").Debugger; }' is not assignable to type 'debug.Debugger & { err: debug.Debugger; }'.
    Property 'diff' is missing in type 'Debugger & { err: Debugger; }' but required in type 'Debugger'.

106     log: Debugger & {
        ~~~

  ../../node_modules/libp2p/node_modules/@types/debug/index.d.ts:47:9
    47         diff: number;
               ~~~~
    'diff' is declared here.
```

since `@types/debug` is resolved to 4.1.7 in libp2p 0.35.6, we should go with it

--- a/package.json
+++ b/package.json
@@ -40,7 +40,7 @@
   },
   "homepage": "https://github.com/ChainSafe/js-libp2p-gossipsub#readme",
   "dependencies": {
-    "@types/debug": "^4.1.5",
+    "@types/debug": "^4.1.7",
     "debug": "^4.3.1",
     "denque": "^1.5.0",
     "err-code": "^3.0.1",

